internal/middleware/middleware.go

```diff
@@ -0,0 +1,31 @@
+package middleware
+
+import (
+	"eventapi/internal/session"
+	"fmt"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/redis/go-redis/v9"
+)
+
+func UseSession(c *fiber.Ctx) error {
+	ss := session.NewSessionService()
+	sId := c.Cookies("session")
+	if sId != "" {
+		sessionData, err := ss.GetSession(c, sId)
+		if err == redis.Nil {
+			ss.ClearSession(c, sId)
+			return c.Redirect("/api/login", 303)
+		}
+
+		if err != nil {
+			fmt.Println("Could not get session info")
```

Comment:
Just going to leave a couple wrap examples:

```go
if err != nil {
  return fmt.Errorf("error retrieving session info: %w", err)
}
```

The error handling middleware could be responsible for logging in this case